### ISSUE TITLE
Ability to nest transfer functions

### ISSUE THREAD

**Issue by arvoelke** [arvoelke/nengolib#84]
i.e., `F(H(s)^{-1})` as a `LinearSystem`

**Comment by arvoelke** [comment]
Once this is done, we can check `F(H(s)^{-1})` to see if the ZOH resulted in unstable poles.

**Comment by arvoelke** [comment]
For example, after mapping onto the alpha, the new poles are simply the old poles duplicated by a reflection about the line `-1 / tau`.

```
theta = 5.0
tau = 0.1
F = nengolib.synapses.PureDelay(theta, 5)
synapse = nengolib.synapses.Alpha(tau)

FH = nengolib.synapses.ss2sim(F, synapse)
Fhat = nengolib.signal.LinearSystem((FH.num(synapse.den), FH.den(synapse.den)))

pylab.figure()
pylab.scatter(Fhat.poles.real, Fhat.poles.imag, marker='x', s=70, lw=2, alpha=0.7, c='blue', label=r'$\hat{F}$')
pylab.scatter(F.poles.real, F.poles.imag, marker='+', s=70, lw=2, alpha=0.7, c='green', label=r'$F$')
pylab.scatter(-2/tau - F.poles.real, F.poles.imag, marker='s', s=50, lw=2, alpha=0.4, c='red', label=r'New')
pylab.vlines([-1/tau], min(Fhat.poles.imag), max(Fhat.poles.imag))
pylab.legend()
pylab.show()
```

![pole_plot](https://cloud.githubusercontent.com/assets/5420057/23042330/3d8da164-f466-11e6-980b-0cf799ccf2de.png)

**Comment by arvoelke** [comment]
Another note to self: the poles of the delay system are proportional to `-1 / theta`, with a constant of proportionality that is positive and depends on which pole and the order of the approximation (and increases as the order increases).

**Comment by celiasmith** [comment]
So basically you need to make sure that the delay (theta) doesn't get too small (moving the poles leftward) compared to tau (which moves poles rightward as it gets bigger).  I suspect making theta larger and tau smaller will be more common, so this doesn't seem likely to be that limiting.

**Comment by arvoelke** [comment]
Yep, exactly. I find this interesting because the converse predicts that, for an alpha with some fixed tau, you need the delay to be at least a certain amount otherwise the poles flip over the line and may even become positive. This is unless the input derivatives are a part of the representation, in which case this is all moot, which is also interesting. :)
